insights/parsers/dmsetup.py

```diff
@@ -121,3 +130,155 @@ def __getitem__(self, idx):
+
+        data = [
+            SetupStatus(
+                device_name='rootvg-tanlv', start_sector='0',
+                num_sectors='6291456', target_type='linear',
+                target_args=None, parsed_args=None,
+            ), ...
+        ]
+
+    Attributes:
+        data (list): List of devices found, in order using SetupStatus structure
+        names (list): Device names, in order found
+        by_name (dict): Access to each device by devicename
+
+    Example:
+        >>> len(dmsetup_status)
+        12
+        >>> dmsetup_status.names[0]
+        'rootvg-tanlv'
+        >>> dmsetup_status[1].target_type
+        'snapshot'
+        >>> dmsetup_status[1].start_sector
+        '0'
+        >>> len(dmsetup_status.by_name)
+        12
+        >>> dmsetup_status.data[-1].parsed_args['used_metadata_blocks']
+        '697'
+        >>> dmsetup_status.data[-1].parsed_args['total_metadata_blocks']
+        '2048'
+        >>> dmsetup_status.data[-1].parsed_args['opts']
+        ['rw', 'no_discard_passdown', 'queue_if_no_space', '-']
+    """
+
+    def parse_content(self, content):
+        self.data = []
+        for line in content:
+            _device_name, _device_info_str = line.rsplit(':', 1)
+            device_name = _device_name.strip()
+            device_info_spl = _device_info_str.strip().split(' ', 3)
+            if len(device_info_spl) < 3:
+                continue
+            target_type = device_info_spl[2]
+            target_args = device_info_spl[3] if len(device_info_spl) == 4 else None
+            parsed_args = None
+            if target_args:
+                parsed_args = self._parse_target_args(target_type, target_args)
+            self.data.append(SetupStatus(
+                device_name=device_name,
+                start_sector=device_info_spl[0],
+                num_sectors=device_info_spl[1],
+                target_type=target_type,
+                target_args=target_args,
+                parsed_args=parsed_args,
+            ))
+        self.names = [dm[0] for dm in self.data]
+        self.by_name = dict((dm[0], dm) for dm in self.data)
+
+    def _parse_target_args(self, target_type, target_args):
+        pars_func_name = '_parse_target_args_' + target_type.replace('-', '_')
+        pars_func = getattr(self, pars_func_name, None)
+        return pars_func(target_args) if pars_func else None
+
+    def _parse_target_args_thin_pool(self, target_args):
+        """
+        Format:
+            <transaction_id> <used_metadata_blocks>/<total_metadata_blocks>
+            <used_data_blocks>/<total_data_blocks> <held_metadata_root>
+            ro|rw|out_of_data_space [no_]discard_passdown [error|queue]_if_no_space
+            needs_check|- metadata_low_watermark
+        Refer to https://www.kernel.org/doc/Documentation/device-mapper/thin-provisioning.txt .
+        """
+        args = target_args.split()
+        if len(args) < 8 or '/' not in args[1] or '/' not in args[2]:
```

Comment:
The same as above: here just skip it w/o any prompt info?

Reply:
Changed to capture such lines into `self.unparseable_lines`.